deepchecks/nlp/dataset.py

```diff
@@ -0,0 +1,557 @@
+    convert_datetime : bool , default: True
+        If set to true, date will be converted to datetime using pandas.to_datetime.
+    datetime_args : t.Optional[t.Dict] , default: None
+        pandas.to_datetime args used for conversion of the datetime column.
+        (look at https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html for more documentation)
+    max_categorical_ratio : float , default: 0.01
+        The max ratio of unique values in a column in order for it to be inferred as a
+        categorical feature.
+    max_categories : int , default: None
+        The maximum number of categories in a column in order for it to be inferred as a categorical
+        feature. if None, uses is_categorical default inference mechanism.
+    label_type : str , default: None
+        Used to assume target model type if not found on model. Values ('classification_label', 'regression_label')
+        If None then label type is inferred from label using is_categorical logic.
+    """
+
+    _dataset: datasets.Dataset
+    _task_type: t.Optional[TaskType]
+    _has_label: bool
+    _label_name: str
+    _is_multilabel: bool = False
+
+    def __init__(
+            self,
+            raw_text: t.Sequence[str],
+            label: t.Optional[TNLPLabel] = None,
+            task_type: str = None,
+            label_name: t.Optional[str] = None,
+    ):
+
+        # Require explicitly setting task type if label is provided
+        if task_type is None:
+            if self.label is not None:
+                raise DeepchecksValueError('task_type must be set when label is provided')
+            self._task_type = None
+        elif task_type == 'text_classification':
+            self._task_type = TaskType.CLASSIFICATION
+        elif task_type == 'token_classification':
+            self._task_type = TaskType.TOKEN_CLASSIFICATION
+        else:
+            raise DeepchecksNotSupportedError(f'task_type {task_type} is not supported, must be one of '
+                                              f'text_classification, token_classification')
+
+        # Set label name to default if not provided
+        if label_name is None:
+            self._label_name = DEFAULT_LABEL_NAME
+        elif isinstance(label_name, (str, float, int)):
+            self._label_name = label_name
+        else:
+            raise DeepchecksNotSupportedError(f'label_name {label_name} is not supported, must be one of '
+                                              f'str, float, int')
+
+        self._validate_text(raw_text)
+        label = self._validate_and_process_label(label, raw_text)
+        # Create dataset
+        self._dataset = datasets.Dataset.from_dict({'text': raw_text, self._label_name: label})
+        self.encode_label()
+
+    def encode_label(self):
+        """
+        Encode label as a ClassLabel object
+        """
+        if self._is_multilabel:
```

Comment:
This argument is not properly initialized in all cases + name is amigieus
Would suggest adding a TaskType of MultiClassClassification (would help down the line)